src/main/java/com/gotocompany/depot/maxcompute/record/RecordDecorator.java

```diff
@@ -0,0 +1,24 @@
+package com.gotocompany.depot.maxcompute.record;
+
+
+import com.gotocompany.depot.maxcompute.model.RecordWrapper;
+import com.gotocompany.depot.message.Message;
+
+import java.io.IOException;
+
+public abstract class RecordDecorator {
+    private final RecordDecorator decorator;
+
+    public RecordDecorator(RecordDecorator decorator) {
+        this.decorator = decorator;
+    }
+
+    public void decorate(RecordWrapper recordWrapper, Message message) throws IOException {
+        append(recordWrapper, message);
+        if (decorator != null) {
+            decorator.decorate(recordWrapper, message);
+        }
+    }
+
+    public abstract void append(RecordWrapper recordWrapper, Message message) throws IOException;
+}
```

Comment:
The decorator chain implementation could lead to StackOverflowError with deep chains. Consider using composition over inheritance and implementing a proper Chain of Responsibility pattern.

Reply:
I see your point about the risk of a StackOverflowError in cases of deep decorator chains. Given our current use case with only two decorators, this recursive approach is simple and sufficient for now. If the number of decorators grows or we anticipate deeper chains in the future, we can refactor to use a proper Chain of Responsibility pattern to address scalability and safety concerns.